CondFormats/HGCalObjects/interface/HGCalESProducerTools.h

```diff
@@ -0,0 +1,97 @@
+#ifndef CondFormats_HGCalObjects_HGCalESProducerTools_h
+#define CondFormats_HGCalObjects_HGCalESProducerTools_h
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <string>
+#include <sstream>    // for std::istringstream
+#include <fnmatch.h>  // to match glob patterns
+#include <nlohmann/json.hpp>
+using json = nlohmann::ordered_json;  // ordered_json preserves key insertion order
+
+namespace hgcal {
+
+  // @short search first match to a given typecode key in a JSON (following insertion order)
+  // allow glob patterns, e.g. 'ML-*', 'M[LH]-[A-Z]3*', etc.
+  inline std::string search_modkey(const std::string& module, const json& data, const std::string& name = "") {
```

Comment:
These functions (except _maybe_ for the last `check_keys()`) would likely be better as out-of-line functions given that they are rather long and have complicated logic.

Reply:
I'll move the main code to `CondFormats/HGCalObjects/src/HGCalESProducerTools.cc`.

Reply:
Done in https://github.com/cms-sw/cmssw/pull/48137/commits/5ba71768546ae894cbffa898b9c8b1bd9cad9a5b.